File: src/matlab/+jtlib/calculateObjectSelectionFeatures.m
function [Area, Solidity, FormFactor] = calculateObjectSelectionFeatures(Objects)
    
    % Measure basic area/shape features
    props = regionprops(logical(Objects), 'Area', 'Solidity', 'Perimeter');

    % Features used for object selection
    Solidity = cat(1, props.Solidity);
    Area = cat(1, props.Area);
    tmp = log((4*pi*cat(1,props.Area)) ./ ((cat(1,props.Perimeter)+1).^2))*(-1);
    tmp(tmp<0) = 0;
    FormFactor = tmp;

end

File: src/python/jtmodules/calculate_object_selection_features.py

# Created on 25-March-2016 by Dr. Saadia Iftikhar, saadia.iftikhar@fmi.ch
# ---------------------------------------------------------------------

from skimage.measure import regionprops
import numpy as np

def calculate_object_selection_features(objects=None,*args,**kwargs):

    solidity = list()
    area = list()
    p = list()
    tmp = list()
    
    for region in regionprops(objects):

        solidity.append(region.solidity)
        area.append(region.area)
        p.append(region.perimeter) 
        n1 = 4 * np.pi * region.area
        n2 = (region.perimeter + 1)
        if ~np.isnan(np.log2((n1/n2))):
            tmp.append(np.log2((n1/n2)))
        else: 
            tmp.append(0)

    tmp[tmp < 0]=0
    formfactor=tmp

    return area,solidity,formfactor
